src/server/zset_family.cc

```diff
@@ -926,6 +960,68 @@ OpResult<StoreArgs> ParseStoreArgs(CmdArgList args) {
   return store_args;
 };
 
+struct UnionArgs {
+  StoreArgs store_args;
+  bool with_scores = false;
+};
+
+OpResult<UnionArgs> ParseUnionArgs(CmdArgList args) {
+  string_view arg_num_keys = "3";
+  StoreArgs store_args;
+  bool with_scores = false;
+
+  if (!absl::SimpleAtoi(arg_num_keys, &store_args.num_keys)) {
+    return OpStatus::SYNTAX_ERR;
+  }
+  DCHECK_GE(args.size(), 1 + store_args.num_keys);
+
+  store_args.weights.resize(store_args.num_keys, 1);
+  for (size_t i = 1 + store_args.num_keys; i < args.size(); ++i) {
+    ToUpper(&args[i]);
+    string_view arg = ArgS(args, i);
+    if (arg == "WEIGHTS") {
+      if (args.size() <= i + store_args.num_keys) {
+        return OpStatus::SYNTAX_ERR;
+      }
+
```

Comment:
The following 40 lines (3 if chains, same structure) are already present in `ParseStoreArgs`. Maybe, we can unify them somehow... Like either add a `ParseUnionArg` function that is called in the loop iteration (that accepts StoreArgs, naming... 🤔 ) or altogether unify those functions into one with options `UnionArgs ParseUnionArgs(..., bool store_kind)`